e2e2/test/cases/netpol/main_test.go

```diff
@@ -0,0 +1,425 @@
+
+func TestMain(m *testing.M) {
+	cfg, err := envconf.NewFromFlags()
+	if err != nil {
+		log.Fatalf("failed to initialize test environment: %v", err)
+	}
+	testenv = env.NewWithConfig(cfg)
+	namespaces := []string{"a", "b", "c"}
+
+	testenv.Setup(
+
+		func(ctx context.Context, config *envconf.Config) (context.Context, error) {
+			client, err := config.NewClient()
+			if err != nil {
+				return ctx, err
+			}
+
+			servers := map[string]string{
+				"a": "a-server",
+				"b": "b-server",
+				"c": "c-server",
+			}
+
+			// 1. Install Latest CNI version
+			log.Print("Install the latest VPC-CNI on the cluster")
+			err = installLatestCNI(config, ctx)
+			if err != nil {
+				log.Print("Failed to install latest aws-vpc-cni on cluster")
+				return ctx, err
+			}
+
+			// 2. Create three namespaces
+			log.Print("Creating the test namespaces")
+			for _, ns := range namespaces {
+				err = createNamespace(ns, client, ctx)
+				if err != nil {
+					log.Printf("Failed to create namespace %s", ns)
+					return ctx, err
+				}
+			}
+
+			// 3. Create deployment and service
+			log.Print("Creating the test deployment and service")
+			for ns, server := range servers {
+				err = createServerAndService(ns, server, 1, client, ctx)
+				if err != nil {
+					log.Printf("Failed to create deployment and service for %s", server)
+					return ctx, err
+				}
+			}
+
+			return ctx, nil
+		},
+	)
+
+	testenv.Finish(
+		func(ctx context.Context, config *envconf.Config) (context.Context, error) {
+			client, err := config.NewClient()
+
+			if err != nil {
+				return ctx, err
+			}
+			log.Print("Deleting the test namespaces")
+			for _, ns := range namespaces {
+				client.Resources().Delete(ctx, &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: ns, Namespace: ns}})
+			}
+
+			return ctx, nil
+		},
+	)
+
+	os.Exit(testenv.Run(m))
+}
+
+func installLatestCNI(config *envconf.Config, ctx context.Context) error {
+
+	repoName := "eks"
+	repoUrl := "https://aws.github.io/eks-charts"
+	chartName := "eks/aws-vpc-cni"
+	releaseNamespace := "kube-system"
+
+	client, err := config.NewClient()
+
+	if err != nil {
+		log.Print("Failed to get client")
+		return err
+	}
+
+	// Delete Cluster role
+	client.Resources().Delete(ctx, &rbac.ClusterRole{ObjectMeta: metav1.ObjectMeta{Name: "aws-node", Namespace: releaseNamespace}})
+
+	// Delete Cluster-Role Binding
+	client.Resources().Delete(ctx, &rbac.ClusterRoleBinding{ObjectMeta: metav1.ObjectMeta{Name: "aws-node", Namespace: releaseNamespace}})
+
+	// Delete ServiceAccount
+	client.Resources().Delete(ctx, &v1.ServiceAccount{ObjectMeta: metav1.ObjectMeta{Name: "aws-node", Namespace: releaseNamespace}})
+
+	// Delete Daemonset
+	client.Resources().Delete(ctx, &appsv1.DaemonSet{ObjectMeta: metav1.ObjectMeta{Name: "aws-node", Namespace: releaseNamespace}})
+
+	// // Delete Configmap
+	client.Resources().Delete(ctx, &v1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Name: "amazon-vpc-cni", Namespace: releaseNamespace}})
```

Comment:
This seems fragile to me, we'll have to keep this in sync with the resources that the managed addon for VPC CNI provides

Reply:
Yes, you are right. I don't think we have an easier way to import the resources into the helm chart at the moment. 
I also considered using eks managed addons to avoid this, but that would mean adding a extra dependency which doesn't seem right here.

I am going to try few more things to see if that helps remove this.
That being said, we have not added new resources to this chart in a while..

Reply:
Found a way to not keep this resource mapping by using `helm template` followed by applying the rendered manifest file. However we would need to use another go library for helm client as the inbuilt helm package (e2e-framework) only validates if the chart can be rendered.